Products block Orderby options

claudiulodro opened the pull request:
Closes #55 

<img width="298" alt="screen shot 2018-03-15 at 11 07 28 am" src="https://user-images.githubusercontent.com/7317227/37482225-165ebed8-2841-11e8-9017-1ae887f2ad82.png">

Disclaimer: The WC API doesn't support ordering by sales, rating, and random and doesn't have filters for modifying it from the feature plugin to support them. We will have to add these features in API v3. Until then the Gutenberg preview won't accurately reflect the orderby setting sometimes, but the frontend will accurately reflect the orderby setting.

warrendholmes commented:
> We will have to add these features in API v3. 

@claudiulodro Does that mean we need to log a new issue on the [repo](https://github.com/woocommerce/wc-api-dev)? I had a look around and couldn't find it anywhere - but that's all quite new to me.

claudiulodro commented:
@warrendholmes I've [opened an issue at the repo](https://github.com/woocommerce/wc-api-dev/issues/98). I don't believe the missing `orderby` options in the API should affect user testing.

I'll have @jameskoster's requested changes today.

claudiulodro commented:
I've removed the random option and added options for ordering by price. The API and the shortcode don't currently support ordering by price, so this functionality will have to be added. I'll add this feature to the shortcode today and I've updated [this issue](https://github.com/woocommerce/wc-api-dev/issues/98) to also document that we need price ordering options.

claudiulodro commented:
[Opened a PR for the new orderby features we need](https://github.com/woocommerce/wc-api-dev/pull/99).

jameskoster commented:
I updated the labels. Hopefully I did everything right :p

claudiulodro commented:
Looks good @jameskoster 👍
